### PR TITLE
GET and POST method

### LINKED ISSUE #8: Adding more header fields to the response for GET
Header fields defined in https://datatracker.ietf.org/doc/html/rfc2616#section-6.2 6.2/7

### PR DESCRIPTION
**Pull request by hoeftint**
Adds the working methods GET and POST to the main branch

### CODE REVIEW

File: interpreters/httpInterpreter.cpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #include "httpInterpreter.hpp"
+#include <iostream>
```
Comment (Finanzcreeper) [inline interpreters/httpInterpreter.cpp]: if this is left over from debugging, please remove it. as it then is an unused header.

File: parsers/ConfigLocationParse.cpp [hunk 2]
```diff
@@ -78,7 +78,7 @@ std::map<std::string, location> parseLocations(std::vector<std::pair<std::string
 		loc.index = parseString(locationChuncks[i], "index");
 		loc.redirect = parseString(locationChuncks[i], "redirect");
 		loc.cgi = parseCgi(locationChuncks[i]);
-		loc.root = getCurrentWorkingDir() + tmp; //can be modified later, according to the structure of the server project
+		loc.root = getCurrentWorkingDir() + parseString(locationChuncks[i], "path") + "/"; //can be modified later, according to the structure of the server project
```
Comment (Finanzcreeper) [inline parsers/ConfigLocationParse.cpp]: is the comment still correct after the change in code?

File: server/MethodExecutor.cpp [hunk 3]
```diff
@@ -1,159 +1,171 @@
...
+#include <string>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <dirent.h>
 
-std::string		root = "/home/thofting/repos/webserv_creeper/content";
 std::string		defaultPage = "index.html";
 
-MethodExecutor::MethodExecutor( void ): _server(0) {};
+MethodExecutor::MethodExecutor( void ): _serverSettings(0) {
```
Comment (Finanzcreeper) [inline server/MethodExecutor.cpp:13]: what is this empty constructor for?

File: server/MethodExecutor.cpp [hunk 4]
```diff
@@ -1,159 +1,171 @@
...
-	_server = server;
-}
+MethodExecutor::MethodExecutor(const t_server *serverSettings): _serverSettings(serverSettings) {}
 
 void	MethodExecutor::wrapperRequest(Request &requ, Response &resp)
 {
-	std::cout << "**** HEADER OF REQUEST: ****\n" << requ.HeaderBuffer << std::endl << "**** END OF HEADER ****" << std::endl;
-	resp.ResponseIntegrity = OK_HTTP;
+	std::cout << "Used route: " + requ.UsedRoute.locationName << std::endl;
+	std::cout << "Used redirect: " + requ.UsedRoute.root << std::endl;
```
Comment (Finanzcreeper) [inline server/MethodExecutor.cpp:21]: There seems to be no iostream inclusion in this file.

File: server/MethodExecutor.cpp [hunk 5]
```diff
@@ -1,159 +1,171 @@
...
-		//	break ;
-		//case (DELETE):
-		//	_executeDelete(requ, resp);
-		//	break ;
-		default:
-			std::cerr << "Method type not found\n";
-			resp.ResponseIntegrity = METHOD_NOT_ALLOWED;
+	resp.headerFields.clear();
+	resp.httpStatus = requ.RequestIntegrity;
+	requ.HeaderFields["content-type"] = "text/plain";
```
Comment (Finanzcreeper) [inline server/MethodExecutor.cpp:30]: is the HeaderFields field "content-type" garantueed to exist? What happens if it doesnt exist?

File: server/MethodExecutor.cpp [hunk 6]
```diff
@@ -1,159 +1,171 @@
...
+
+	if (resp.httpStatus == MOVED_PERMANENTLY)
+		resp.headerFields["location"] = requ.UsedRoute.redirect;
```
Comment (Finanzcreeper) [inline server/MethodExecutor.cpp:33]: Is the HeaderFields field "location" garantueed to exist? What happens if it doesnt exist?

File: server/MethodExecutor.cpp [hunk 7]
```diff
@@ -1,159 +1,171 @@
...
+	// Append body to response
+	if (requ.ReqType != HEAD)
+		resp.responseBuffer.append(resp.body);
 	resp.isReady = true;
-	std::cout << "**** RESPONSE: ****\n" << resp.responseBuffer << "**** END OF RESPONSE ****" << std::endl;
+	//std::cout << "**** RESPONSE: ****\n" << resp.responseBuffer << "**** END OF RESPONSE ****" << std::endl;
 }
 
-void    MethodExecutor::_executeGet(Request &requ, Response &resp)
+void	MethodExecutor::_executePost(Request &requ, Response *resp)
```
Comment (Finanzcreeper) [inline server/MethodExecutor.cpp]: why is resp passed by pointer?

File: server/MethodExecutor.cpp [hunk 8]
```diff
@@ -1,159 +1,171 @@
...
-	
-	std::cout << "PATH:" << path << "*" << std::endl;
-	std::ifstream	ifs(path.c_str());
-	
-	// TODO: More detailed access check for files
-	// - Check if file exists
-	if (ifs.good())
+	std::string	path = requ.RoutedPath;
+	struct stat	s;
+	std::cout << "HALLO" << std::endl;
```
Comment (Finanzcreeper) [inline server/MethodExecutor.cpp]: is this leftover debugging, or is this  left in intentionally?

File: parsers/ConfigLocationParse.cpp (hunk 2 above)
Comment (hoeftint) [reply parsers/ConfigLocationParse.cpp]: I removed the comment. Can be still modified later though. For example if we decide that the path given in any location block should be absolute and not relative to the working dir

File: server/MethodExecutor.cpp (hunk 3 above)
Comment (hoeftint) [reply server/MethodExecutor.cpp:13]: Its not needed, but maybe I want to implement the class in orthodox canonical form later. I leave the left the default constructor in the code

File: server/MethodExecutor.cpp (hunk 5 above)
Comment (hoeftint) [reply server/MethodExecutor.cpp:30]: This is not final and was only done to make an initial implementation of the PULL method. I will handle that in the future but have it in mind

File: server/MethodExecutor.cpp (hunk 6 above)
Comment (hoeftint) [reply server/MethodExecutor.cpp:33]: Location does not need to exist. Nothing bad should happen

File: server/MethodExecutor.cpp (hunk 8 above)
Comment (hoeftint) [reply server/MethodExecutor.cpp]: deleted

**Review by Finanzcreeper — APPROVED** [review]

**Comment by Finanzcreeper** [comment]
I thought that approving the changes only approves the current changes made. if you wanted to change any other thing as well. please do.

File: server/MethodExecutor.cpp (hunk 7 above)
Comment (hoeftint) [reply server/MethodExecutor.cpp]: I dont remember why I did it like that :D. Changed it to reference

**Review by Finanzcreeper — APPROVED** [review]

**Pull request merged**